python-checks/src/main/java/org/sonar/python/checks/UndefinedNameAllPropertyCheck.java

```diff
@@ -149,14 +160,53 @@ private static StringLiteral retrieveStringLiteral(Tree tree) {
     return (StringLiteral) Expressions.singleAssignedValue((Name) tree);
   }
 
-  private static class UnresolvedWildcardImportVisitor extends BaseTreeVisitor {
+  private static boolean importsManipulatedAllProperty(FileInput fileInput) {
+    return fileInput.globalVariables().stream().anyMatch(s -> s.name().equals("__all__") && s.usages().stream().anyMatch(u -> u.kind().equals(Usage.Kind.IMPORT)));
```

Comment:
_nitpick_ I would prefer to avoid nesting `anyMatch` and would rather have a loop for the global variables, but it's just a personal preference, feel free to disregard.